waku/v2/api/publish/message_check.go

```diff
@@ -0,0 +1,198 @@
+package publish
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+	"github.com/waku-org/go-waku/waku/v2/protocol/store"
+	"github.com/waku-org/go-waku/waku/v2/timesource"
+	"go.uber.org/zap"
+)
+
+const maxHashQueryLength = 100
+const hashQueryInterval = 3 * time.Second
+const messageSentPeriod = 3    // in seconds
+const messageExpiredPerid = 10 // in seconds
+
+type MessageSentCheck struct {
```

Comment:
can we rename to something like SentMsgConfirmer or something in that line?

Reply:
Is it a convention to use something like `confirmer` or `checker`?
I find the name `MessageSentCheck` not worse than `SentMsgConfirmer` to express its logic actually.

Reply:
Nothing like that, just that it is a routine performing periodic task of checking/confirming whether messages are sent. In general i have seen such terms being used when routines are involved. 
No hard opinion, just to make it more reader friendly.